### PR TITLE
improve performance of bubbleINfo

### PR DESCRIPTION
**Pull request by pinzart90**
DYN-6438 - long time to run Graph when you disconnect a node (i.e when nodes have errors/warnings)
Add “collapsed” visibility for all bubble controls when no data binding exists.
Reduce thread switching and update all node bubbles at the end of graph to xecution

Without fix, Graph Run takes ~ 1 min.
With fix, Graph Run takes ~ 3 seconds.

### CODE REVIEW

**Comment by mjkkirschner** [comment]
I'm trying to understand how these changes improve things - are they similar to this?
https://github.com/DynamoDS/Dynamo/wiki/Dynamo-UI-Performance#wpf-dispatcher-issues

**Comment by pinzart90** [comment]
> I'm trying to understand how these changes improve things - are they similar to this?
> https://github.com/DynamoDS/Dynamo/wiki/Dynamo-UI-Performance#wpf-dispatcher-issues

Yes. One part of the slowdown was caused by that. Another was that the bubble UI was rendering components that were binding to properties without data context.

File: src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs [hunk 1]
```diff
@@ -698,7 +698,11 @@ private void OnUpdateGraphCompleted(AsyncTask task)
                 var node = workspace.Nodes.FirstOrDefault(n => n.GUID == guid);
                 if (node == null)
                     continue;
+
+                // Block Infos updates during the many errors/warnings/notifications added here
+                // InfoBubbles will be updated on NodeViewModel's EvaluationCompleted handler.
                 using (node.PropertyChangeManager.SetPropsToSuppress(nameof(NodeModel.Infos), nameof(NodeModel.State)))
+                using (Disposable.Create(() => { node.BlockInfoBubbleUpdates = true; }, () => { node.BlockInfoBubbleUpdates = false; }))
```
Comment (aparajit-pratap) [inline src/DynamoCore/Graph/Workspaces/HomeWorkspaceModel.cs:705]: I haven't refreshed my memory on the `PropertyChangeManager` and how it suppresses property-changed events, but my question is are both this and the `blockInfoBubbleUpdates` required? How are these different if both ensure the UI updates on the info bubbles happen at the end of the graph execution? In other words, aren't the updates batched for the end of the graph execution already (with the introduction of `PropertyChangeManager`?

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs [hunk 2]
```diff
@@ -1464,7 +1465,15 @@ private void DisposeErrorBubble()
 
         public void UpdateBubbleContent()
         {
-            if (DynamoViewModel == null) return;
+            if (NodeModel.BlockInfoBubbleUpdates)
```
Comment (RobertGlobant20) [inline src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:1468]: You can simplify this with:
 if (NodeModel.BlockInfoBubbleUpdates == true || DynamoViewModel == null) return;

**Review by RobertGlobant20 — APPROVED** [review]
LGTM, did you find a way to measure the performance improvement done by this fix?

**Comment by pinzart90** [comment]
> LGTM, did you find a way to measure the performance improvement done by this fix?

Manually measured. I could not find any reliable automated way to measure UI responsiveness

File: src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs (hunk 2 above)
Comment (pinzart90) [reply src/DynamoCoreWpf/ViewModels/Core/NodeViewModel.cs:1468]: I find it easier to read this way.

**Review by aparajit-pratap — APPROVED** [review]

**Comment by RobertGlobant20** [comment]
> > LGTM, did you find a way to measure the performance improvement done by this fix?
> 
> Manually measured. I could not find any reliable automated way to measure UI responsiveness

Well... I was asking because I haven't found any reliable way to measure Dynamo graph loading performance, When the performance improvement is of seconds if I use dotTrace I get similar times with/without the fix :(

**Comment by pinzart90** [comment]
https://master-5.jenkins.autodesk.com/job/Dynamo/job/DynamoSelfServe/job/pullRequestValidation/15905
Only one failure, as in master (the remove PII test)

**Pull request merged**